### ISSUE TITLE
Callback with audio frame containing wake word?

### ISSUE THREAD

**Issue by alopix** [Picovoice/Porcupine#26]
Is there a way to get the detected audio in the callback?
It would be really nice to have the callback include the audio frames so e.g. cloud-based wake word verification for the Alexa Voice Service can be used.

**Comment by kenarsa** [comment]
That's a great idea. Actually, we are in the process of integrating Porcupine with AVS SDK. They have their own interface for wake-word engines. I wonder if it possible there? I will let you know when we get there.

Do you have any other use case other than AVS cloud verification? If yes, then let us know and we try to put it on our TODO list. Otherwise, a quick hack that can unblock you could be to have a circular buffer of audio for past 2 seconds or so and whenever the keyword is detected send that to the cloud. Thoughts?

**Comment by kenarsa** [comment]
closing as there is no activity.
